src/nvim/main.c

```diff
@@ -849,9 +850,11 @@ static void command_line_scan(mparm_T *parmp)
           } else if (STRICMP(argv[0] + argv_idx, "embed") == 0) {
             embedded_mode = true;
             const char *err;
-            if (!channel_from_stdio(true, CALLBACK_READER_INIT, &err)) {
+            if (!did_embed
```

Comment:
We could also do `if (embedded_mode) { channel_from_stdio(...) }` in main.c, which would be more consistent with "listen" for example.

Reply:
You mean just ignore the error from `channel_from_stdio()` here instead of tracking whether we've set it up?